src/XMakeBuildEngine/BackEnd/Components/Communications/NodeProviderInProc.cs

```diff
@@ -230,7 +230,11 @@ public bool CreateNode(int nodeId, INodePacketFactory factory, NodeConfiguration
 
                 if (_exclusiveOperatingEnvironment)
                 {
+#if FEATURE_NAMED_SYNCHRONIZATION_PRIMITIVES
                     _inProcNodeOwningOperatingEnvironment = new Semaphore(1, 1, "MSBuildIPN_" + Process.GetCurrentProcess().Id);
+#else
+                    _inProcNodeOwningOperatingEnvironment = new Semaphore(1, 1, null);
```

Comment:
@rainersigwald, I have changed it to use `static` semaphore and rebased.
The old diff is at https://github.com/jasonwilliams200OK/msbuild/commit/e7c17e4db5a27c2ad2d3e6614e9a0d6256e6092f.